### PR TITLE
NxAlert RTL Tests - RSC-1223

### PR DESCRIPTION
**Pull request by rpokorny**
https://issues.sonatype.org/browse/RSC-1223

### CODE REVIEW

File: lib/src/components/NxAlert/__tests__/NxAlert.test.tsx [hunk 1]
```diff
@@ -4,263 +4,158 @@
...
-        component = getNxAlert({ onClose });
+  const minimalProps = {
+        children: 'A message to show in an alert',
+        icon: faBiohazard
+      },
+      quickRender = rtlRender(NxAlert, minimalProps);
 
-    expect(component).toContainMatchingElement(NxCloseButton);
-    expect(onClose).not.toHaveBeenCalled();
+  baseTests(NxAlert, minimalProps);
```
Comment (EmilyBermudez88) [inline lib/src/components/NxAlert/__tests__/NxAlert.test.tsx:80]: does the generic NxAlert need to have tests for role assignment as well, or is it considered sufficient when its tested in all the variations of NxAlert?

File: lib/src/components/NxAlert/__tests__/NxAlert.test.tsx [hunk 2]
```diff
@@ -4,263 +4,158 @@
...
+        expect(viewWithRole.container.firstElementChild).toHaveAttribute('role', 'foo');
       });
 
-      it('passes any other props to the NxAlert', function() {
-        const onClose = jest.fn(),
-            component = getNxInfoAlert({ id: 'foo', title: 'baz', onClose });
-        expect(component).toHaveProp('id', 'foo');
-        expect(component).toHaveProp('title', 'baz');
-        expect(component).toHaveProp('onClose', onClose);
+      it('renders an img with an accessible nameof "Warning"', function() {
```
Comment (EmilyBermudez88) [inline lib/src/components/NxAlert/__tests__/NxAlert.test.tsx:141]:
```suggestion
      it('renders an img with an accessible name of "Warning"', function() {
```

Comment (EmilyBermudez88) [reply lib/src/components/NxAlert/__tests__/NxAlert.test.tsx:141]: (in all `it('renders an img with an accessible nameof...)`

File: lib/src/components/NxAlert/__tests__/NxAlert.test.tsx (hunk 1 above)
Comment (rpokorny) [reply lib/src/components/NxAlert/__tests__/NxAlert.test.tsx:85]: > does the generic NxAlert need to have tests for role assignment as well, or is it considered sufficient when its tested in all the variations of NxAlert?

This made me realize there was room for several improvements around role testing.  All of the different alert variations _can_ receive a role, and the ones that don't provide a role by default should probably have tests proving that.  Addressed in 72da7ca06

File: lib/src/components/NxAlert/__tests__/NxAlert.test.tsx (hunk 2 above)
Comment (rpokorny) [reply lib/src/components/NxAlert/__tests__/NxAlert.test.tsx]: 3c5b1c0a2

**Review by EmilyBermudez88 — APPROVED** [review]

**Pull request merged**